### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by rabbah**
Fix couple of bugs in JsHelpers where there was an assumption that nested values are instances of JsObject. Move helper to whisk.common and make trait a singleton instead.

Also prevent logging of "malformed credentials" when there are none presented.
PG2/881.

### CODE REVIEW

File: core/controller/src/main/scala/whisk/core/controller/Authenticate.scala [hunk 1]
```diff
@@ -72,7 +72,7 @@ trait Authenticate extends Logging {
                 future
             }.toOption
         } getOrElse {
-            info(this, s"credentials are malformed")
+            userpass.map(_ => info(this, s"credentials are malformed"))
```
Comment (psuter) [inline core/controller/src/main/scala/whisk/core/controller/Authenticate.scala]: surely you meant to write `foreach`

File: common/scala/src/main/scala/whisk/utils/JsHelpers.scala [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import spray.json.JsValue
+
+object JsHelpers {
+    implicit class JsObjectHelper(js: JsObject) {
+        def getFieldPath(path: List[String]): Option[JsValue] = {
+            path match {
+                case Nil       => Option(js)
+                case p :: Nil  => js.fields.get(p)
+                case p :: tail => js.fields.get(p) match {
+                    case Some(o: JsObject) => o.getFieldPath(tail)
```
Comment (psuter) [inline common/scala/src/main/scala/whisk/utils/JsHelpers.scala]: Probably not terribly efficient given that every recursive call involves creating a new instance of the `implicit` class

Comment (rabbah) [inline common/scala/src/main/scala/whisk/utils/JsHelpers.scala]: https://github.com/openwhisk/openwhisk/pull/1720/commits/c1db4e5fb67ede51ded62b89910c6de12f2b73d8

File: core/controller/src/main/scala/whisk/core/controller/Authenticate.scala (hunk 1 above)
Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/controller/Authenticate.scala]: thanks for the nudge.

**Comment by rabbah** [comment]
rebased to master to resolve conflicts.

**Comment by psuter** [comment]
Merged as 121d63149cc6d6f7c9cf4f85e5150cde537ab413...ad9b444268068a9d60fcaf109fd00ac8cee8dfc5

**Pull request closed without merging**